### PR TITLE
add support for multiplying parentheses

### PR DESCRIPTION
**Pull request by juca1331**
Hello, the package didn't have the capability to perform mathematical operations with parentheses, so I decided to add this functionality within the parse method of the Parser class. To enable this functionality, it's necessary to set the 'multiplyWithParentheses' attribute to true in the parse method.

### CODE REVIEW

**Comment by fkleon** [comment]
Hi @juca1331, thanks for your pull request!

If I'm understand this change correctly, it will allow the parser to interpret a missing operator between parenthesis to be an implicit multiplication? To be honest, the code is a bit hard to follow. If that understanding is correct, I'm not sure that string manipulation on the input expression string is the best way to achieve this. Such functionality would be better implemented during or after the tokenization step within the parser.

A few further notes:
* Could you please update this PR to only include relevant changes? There is quite a lot of noise, but I think the relevant  files are only `parser.dart` and the CHANGELOG file.
* Any new functionality requires unit tests to be added. This proves that the code is correct, ensures it is maintainable going forward, and also helps readers to understand how the feature is supposed to work.

**Comment by juca1331** [comment]
I already made all the changes that you ask me to do, to merge the pull request  :)  🥇

**Comment by fkleon** [comment]
@juca1331 Thanks for the update! Implementing this in the Lexer is a much better approach.

* I noticed there are still a bunch of unrelated files that you have committed, e.g. all of the `build/` folder. Can you please remove those from the pull request?
* The linter is not happy about some of the formatting. You should be able to simply run `dart format .` to fix those.

**Comment by juca1331** [comment]
I've already done all the changes you asked for :)

**Review by fkleon — APPROVED** [review]
Thank you, this looks good. I'll merge this but will make some follow-up changes to tweak how the parser can be configured. I'd like to use options when constructing the instance, not when calling the tokenize methods.
